include/zephyr/drivers/i3c.h

```diff
@@ -0,0 +1,2180 @@
+ * <tt>I3C_DEVICE_DESC_DT_WITH_IBI_PAYLOAD(DT_DRV_INST(inst), ibi_cb)</tt>.
+ *
+ * @param inst Devicetree instance number
+ * @param ibi_cb Pointer to IBI callback.
+ */
+#define I3C_DEVICE_DESC_DT_INST_WITH_IBI_CB(inst, ibi_cb)		\
+	I3C_DEVICE_DESC_DT_WITH_IBI_CB(DT_DRV_INST(inst, ibi_cb))
+
+/**
+ * @brief Structure initializer for i3c_device_desc from devicetree instance
+ *
+ * This is equivalent to
+ * <tt>I3C_DEVICE_DESC_DT(DT_DRV_INST(inst))</tt>.
+ *
+ * @param inst Devicetree instance number
+ */
+#define I3C_DEVICE_DESC_DT_INST(inst) \
+	I3C_DEVICE_DESC_DT(DT_DRV_INST(inst))
+
+/**
+ * @brief Like DEVICE_DT_DEFINE() with I3C target device specifics.
+ *
+ * Defines a I3C target device which implements the I3C target device API.
+ *
+ * @param node_id The devicetree node identifier.
+ *
+ * @param init_fn Name of the init function of the driver.
+ *
+ * @param pm_device PM device resources reference (NULL if device does not use PM).
+ *
+ * @param data_ptr Pointer to the device's private data.
+ *
+ * @param cfg_ptr The address to the structure containing the
+ *                configuration information for this instance of the driver.
+ *
+ * @param level The initialization level. See SYS_INIT() for
+ *              details.
+ *
+ * @param prio Priority within the selected initialization level. See
+ *             SYS_INIT() for details.
+ *
+ * @param api_ptr Provides an initial pointer to the API function struct
+ *                used by the driver. Can be NULL.
+ */
+#define I3C_DEVICE_DT_DEFINE(node_id, init_fn, pm_device,		\
+			     data_ptr, cfg_ptr, level, prio,		\
+			     api_ptr, ...)				\
+	DEVICE_DT_DEFINE(node_id, init_fn, pm_device,			\
+			 data_ptr, cfg_ptr, level, prio,		\
+			 api_ptr, __VA_ARGS__)
+
+/**
+ * @brief Like I3C_TARGET_DT_DEFINE() for an instance of a DT_DRV_COMPAT compatible
+ *
+ * @param inst instance number. This is replaced by
+ * <tt>DT_DRV_COMPAT(inst)</tt> in the call to I3C_TARGET_DT_DEFINE().
+ *
+ * @param ... other parameters as expected by I3C_TARGET_DT_DEFINE().
+ */
+#define I3C_DEVICE_DT_INST_DEFINE(inst, ...)				\
+	I3C_DEVICE_DT_DEFINE(DT_DRV_INST(inst), __VA_ARGS__)
+
+/**
+ * @brief Structure describing a I2C device on I3C bus.
+ *
+ * Instances of this are passed to the I3C controller device APIs,
+ * for example:
+ * () i3c_i2c_device_register() to tell the controller of an I2C device.
+ * () i3c_i2c_transfers() to initiate data transfers between controller
+ *    and I2C device.
+ *
+ * Fields other than @c node must be initialized by the module that
+ * implements the device behavior prior to passing the object
+ * reference to I3C controller device APIs.
+ */
+struct i3c_i2c_device_desc {
+	/** Private, do not modify */
+	sys_snode_t node;
+
+	/** I3C bus to which this I2C device is attached */
+	const struct device *bus;
+
+	/** Static address for this I2C device. */
+	const uint8_t addr;
+
+	/**
+	 * Legacy Virtual Register (LVR)
+	 * - LVR[31:8]: Unused.
```

Comment:
Where are these unused bits coming from? I'm not seeing anything about unused [31:8] defined the spec

Reply:
Ahh... I copied it from DTS binding file where over there, each reg element is 32-bit integer, and that's why I added the unused part. Will remove.